trapper/commands.py

```diff
@@ -94,20 +104,172 @@ def add_subparser(
         return subparser
 
 
+@Subcommand.register("run-distributed")
+class DistributedRun(Subcommand):
+    """Start an experiment in a distributed data parallel fashion using the config file.
+    It utilizes pytorch's elastic run under the hood. To see supported features see
+    https://pytorch.org/docs/stable/elastic/run.html
+
+    Usage:
+        Basic:
+            ` trapper run-distributed experiment.jsonnet <--other-kwargs vals>`
+    """
+
+    def add_subparser(
+        self, parser: argparse._SubParsersAction
+    ) -> argparse.ArgumentParser:
+        description = """Run the experiment specified in the config file in a 
+            distributed fashion across multiple devices (DDP)."""
+        subparser = parser.add_parser(
+            self.name,
+            description=description,
+            help="Start distributed training of a model.",
+        )
+
+        subparser.add_argument(
+            "config_path",
+            type=str,
+            help="path to the experiment config file in `json` or `jsonnet`"
+            " format describing the model, dataset and other details.",
+        )
+
+        subparser.add_argument(
+            "-o",
+            "--overrides",
+            type=str,
+            default="",
+            help=(
+                "a json or jsonnet structure used to override the experiment "
+                "configuration, e.g., '{\"optimizer.lr\": 1e-5}'.  Nested "
+                "parameters can be specified either with nested dictionaries "
+                "or with dot syntax."
+            ),
+        )
+
+        subparser.set_defaults(func=run_distributed)
+
+        return subparser
+
+
+@record
+def run_distributed(args=None):
+    """
+    Taken from
+    https://github.com/pytorch/pytorch/blob/4618371da56c887195e2e1d16dad2b9686302800/torch/distributed/run.py
+    and modified to properly run by trapper.
+    """
+
+    def merge_json_str(opts: List[str]) -> None:
```

Comment:
I think this function needs an explanation. Specifically why is it needed, the details on arguments.

Reply:
I added a note section in docstring.